symbolic shapes: add parenthesis around FloorDiv expression

nunoplopes opened the pull request:
Before it would print the guard expression like:
`2*3//2`
and now:
`2*(3//2)`

```python
print(2*3//2)   # 3
print(2*(3//2)) # 2
```

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire @ezyang

ezyang reviewed (approved):
whoops. maybe we should test this 🤔

ezyang commented:
it looks like this wobbled some numbers and so they will have to be updated. Do you need help updating them?

ezyang commented:
Also, it's kind of weird that the enclosing operator (in this case, multiply) isn't handling the parenthesization itself. Maybe we didn't register floor divs precedence correctly? 🤔 

cc @asmeurer

nunoplopes commented:
You were right; precedence wasn't set for the expression.
I think we now get the expected result:
```python
shape_env = ShapeEnv()
a = create_symint(shape_env, 2)
b = create_symint(shape_env, 3)
c = create_symint(shape_env, 4)
d = create_symint(shape_env, 5)
expr = ((a + b) // (c * d)) * a + (a//b)**2 > d
bool(expr)
print(shape_env.format_guards())
#  - s0*((s0 + s1)//s2*s3) + (s0//s1)**2 <= s3
```

Let me try to find a machine to build PyTorch to fix the tests..

ezyang commented on torch/fx/experimental/symbolic_shapes.py:
```diff
@@ -217,17 +217,13 @@ class FloorDiv(sympy.Function):
         2. Printing out the expression is nicer (compared to say, representing a//b as (a - a % b) / b)
         """
         nargs = (2,)
+        precedence = 49  # precedence of mul - 1  # noqa: F811
```
Shouldn't floordiv have the same precedence as mul, per Python's operator precedence rules?

nunoplopes replied:
That works, thanks. The printer handles commutativity conservatively.
Fixed.

nunoplopes committed:
symbolic shapes: add parenthesis around FloorDiv expression

Before it would print the expression like:
2*3//2
and now:
2*(3//2)

print(2*3//2) is 3
print(2*(3//2)) is 2

ezyang commented:
@pytorchbot merge -r master

ezyang commented:
@pytorchbot merge -f "only failure on disabled tests"
